### PR TITLE
System Roles should consistently use ansible_managed in configuration files it manages

### PR DESCRIPTION
**Pull request by nhosoi**
bz#2044640

The certificate role needs to generate the hook scripts with the
correct `ansible_managed` string commented in the same format as
generated by the template module.  Borrowed the method from the
kernel_settings to get the ansible managed comment and updated
the base class in module_utils so that it adds the comment to
the pre and post scripts as follows.
```
  ==> {pre,post}-scripts/<filename>.sh <==
  #!/bin/bash
  #
  # Ansible managed
  #

  ... script ...
```

### CODE REVIEW

**Review by rjeffman — APPROVED** [review]
lgtm.

**Comment by nhosoi** [comment]
> lgtm.

Thank you for your reviewing the change, @rjeffman !!
As I do not have the privilege, could you please merge this pr? Thanks!

File: module_utils/certificate_lsr/providers/base.py [hunk 1]
```diff
@@ -672,6 +676,14 @@ def _write_param_to_file_if_diff(self, param_name, filepath, check_mode):
...
         # if file and param are the same just return
         if param_sha1 == file_sha1:
+            # Even if file and param are identical,
+            # if the script exists and the content of the script
+            # is different from the current one including the
+            # ansible managed comment, update the script.
+            cur_script = open(filepath, encoding="utf-8").read()
+            if param != cur_script:
+                with open(filepath, "w") as script_fp:
+                    script_fp.write(param)
```
Comment (richm) [inline module_utils/certificate_lsr/providers/base.py:686]: What happens if the only change made is a change in the ansible_managed value?  Will the file be updated, and the module report `changed: true`?

File: library/certificate_request.py [hunk 2]
```diff
@@ -172,6 +179,7 @@
     name: single-example
     dns: www.example.com
     ca: self-sign
+    ansible_managed_new: #\n# Ansible managed\n#\n
```
[the remark this answers is not in the document]
Comment (nhosoi) [reply library/certificate_request.py:182]: Even if we define the variable with `required: true`, it's ok not to have it in the example? Or should we define it with `required: false`?

File: module_utils/certificate_lsr/providers/base.py (hunk 1 above)
Comment (nhosoi) [reply module_utils/certificate_lsr/providers/base.py:686]: You are right, @richm. I should have return `True` here. Thanks!

File: library/certificate_request.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (richm) [reply library/certificate_request.py:182]: > Even if we define the variable with `required: true`, it's ok not to have it in the example?

Yes.  Users of the certificate role should read the *role* documentation, not the *module* documentation, and use examples from the *role* README or the `examples/` directory (if any).

> Or should we define it with `required: false`?

No, it should be required.

**Comment by nhosoi** [comment]
@richm, @rjeffman , could you please review this pr one more time? Thanks!!

**Comment by richm** [comment]
> @richm, @rjeffman , could you please review this pr one more time? Thanks!!

lgtm, but I want to make sure the ansible-2.12 tests, or at least some of them, pass - right now the network PRs are using all of the workers :P

**Review by rjeffman — APPROVED** [review]
lgtm.

**Comment by tunsira** [comment]
Open

**Comment by richm** [comment]
> Open

Yes?

**Pull request merged**